testsuites/benchmark/src/main/java/org/eclipse/rdf4j/benchmark/rio/util/DataSetGenerator.java

```diff
@@ -0,0 +1,430 @@
+	private ArrayList<IRI> predicateIRIs;
+	private ArrayList<IRI> contextIRIs;
+	private Random randomResource;
+	private Random randomProbability;
+	private BNode prevBNode;
+	private IRI currentContext;
+	private int totalWords;
+	private int avgWordSize;
+
+	public DataSetGenerator() {
+		this.factory = SimpleValueFactory.getInstance();
+		this.valueBuilder = new StringBuilder();
+		this.randomResource = new Random(64);
+		this.randomProbability = new Random(64);
+		this.avgWordSize = 0;
+		fetchWords();
+		generatePredicateList();
+	}
+
+	public File generateStatementsFile(RDFFormat format, String fileName, int percentBNodes, int percentLiterals,
+			int textMinLength, int textMaxLength, int totalStatements, boolean isTextOnly,
+			boolean withContext) throws IOException {
+		assert percentBNodes + percentLiterals <= 100 && totalStatements > 0 && textMaxLength > 0
+				&& textMinLength < textMaxLength;
+		File file = createFile(format, fileName);
+		if (format.equals(RDFFormat.BINARY)) {
+			FileOutputStream out = new FileOutputStream(file);
+			if (withContext) {
+				generateContextList(totalStatements);
+				writeBinaryStatementsWithContext(out, percentBNodes, percentLiterals, textMinLength, textMaxLength,
+						totalStatements, isTextOnly);
+				maxCurrentContextCount.set(0);
+				contextCount.set(0);
+				currentContext = null;
+			} else {
+				writeBinaryStatementsWithoutContext(out, percentBNodes, percentLiterals, textMinLength,
+						textMaxLength,
+						totalStatements, isTextOnly);
+			}
+		} else {
+			BufferedWriter out = new BufferedWriter(
+					new OutputStreamWriter(new FileOutputStream(file), StandardCharsets.UTF_8));
+			if (withContext) {
+				generateContextList(totalStatements);
+				writeStatementsWithContext(out, format, percentBNodes, percentLiterals, textMinLength, textMaxLength,
+						totalStatements, isTextOnly);
+				maxCurrentContextCount.set(0);
+				contextCount.set(0);
+				currentContext = null;
+			} else {
+				writeStatementsWithoutContext(out, format, percentBNodes, percentLiterals, textMinLength,
+						textMaxLength,
+						totalStatements, isTextOnly);
+			}
+		}
+		prevBNode = null;
+		randomResource = new Random(64);
+		randomProbability = new Random(64);
+		return file;
+	}
+
+	public boolean statementFileExists(String fileName, RDFFormat format) {
+		Path path = Paths.get(this.getClass().getProtectionDomain().getCodeSource().getLocation().getPath());
+		path = path.getParent().getParent();
+		File file = new File(path.toString(), DIR + fileName + "." + format.getDefaultFileExtension());
+		return file.exists();
+	}
+
+	private File createFile(RDFFormat format, String fileName) throws IOException {
+		Path path = Paths.get(this.getClass().getProtectionDomain().getCodeSource().getLocation().getPath());
+		path = path.getParent().getParent();
+		File file = new File(path.toString(), DIR + fileName + "." + format.getDefaultFileExtension());
+		assert file.createNewFile();
+		return file;
+	}
```

Comment:
Where does this create the file? Would it be better to use `Strings.append(File.separator).to(System.getProperty("java.io.tmpdir"))` ?